HPCC-12131 Xref Upgrade

miguelvazq opened the pull request:
Incorporate Xref into the new ecl watch paradigm.

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexis.com

miguelvazq commented:
@GordonSmith Please review.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 00d1e94190f0bd84329dbf870aa1a72b08145ecd
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: 25fd2c7e14398eeb734c12f0975a4b2ed1f432ef
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/WsDFUXref.js:
```diff
@@ -0,0 +1,52 @@
+/*##############################################################################
+#    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
Year?

GordonSmith replied:
Actually I believe we don't need this copyright message in each file any more as the single licence.txt should suffice?

richardkchapman replied:
@fvillanustre Can you comment on what our policy on copyright messages in files needs to be?

GordonSmith commented on esp/src/eclwatch/XrefDirectoriesWidget.js:
```diff
@@ -0,0 +1,139 @@
+        idProperty: "Name",
+
+        init: function (params) {
+            var context = this;
```
Context not used (I will be enabling this test with the linter soon!)

GordonSmith commented on esp/src/eclwatch/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,124 @@
+        idProperty: "Name",
+
+        init: function (params) {
+            var context = this;
```
context not used

GordonSmith commented on esp/src/eclwatch/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,124 @@
+        createGrid: function (domID) {
+            this.openButton = registry.byId(this.id + "Open");
+
+            var context = this;
```
context not used

GordonSmith commented on esp/src/eclwatch/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,124 @@
+                   newRows.push({
+                        File: row.File,
+                        Text: row.Text,
+                        Status: context.i18n.Warning,
```
trailing comma

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -0,0 +1,101 @@
+        idProperty: "Name",
+
+        init: function (params) {
+            var context = this;
```
context not used

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -0,0 +1,101 @@
+        },
+
+        createGrid: function (domID) {
+            var context = this;
```
context not used

GordonSmith commented on esp/src/eclwatch/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,113 @@
+        idProperty: "Name",
+
+        init: function (params) {
+            var context = this;
```
context not used

GordonSmith commented on esp/src/eclwatch/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,113 @@
+
+        createGrid: function (domID) {
+
+            var context = this;
```
context not used.

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -0,0 +1,133 @@
+        idProperty: "Name",
+
+        init: function (params) {
+            var context = this;
```
context not used.

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -0,0 +1,133 @@
+                id: this.id + "Delete",
+                disabled: false,
+                onClick: function (val) {
+                    if (confirm(context.i18n.RunningServerStrain)) {
```
context not declared.

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -0,0 +1,133 @@
+                            if (response) {
+                                context.refreshGrid();
+                            }
+                        })
```
missing ;

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -0,0 +1,133 @@
+                label: this.i18n.Delete
+            }).placeAt(this.openButton.domNode, "after");
+
+            var context = this;
```
Ahh here it is  - should move to top of procedure.

GordonSmith commented on esp/src/eclwatch/XrefQueryWidget.js:
```diff
@@ -0,0 +1,217 @@
+        idProperty: "Name",
+
+        init: function (params) {
+            var context = this;
```
context not used

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -48,6 +48,8 @@ define({root:
     Bottom: "Bottom",
     Cancel: "Cancel",
+    CancelAll: "Cancel All",
+    CancelAllMessage: "This will attempt to stop any running jobs and clear the job queue. Do you wish to continue?",
```
How about "Abort running jobs and clear queue.  Do you wish to continue?"

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -95,6 +97,8 @@ define({root:
     Delete: "Delete",
     Deleted: "Deleted",
+    DeleteEmptyDirectories: "Delete empty directories?",
+    DeleteDirectories: "You are about to delete empty directories. Do you wish to continue?",
```
"Remove empty directories.  Do you with to continue?"

JamesDeFabia replied:
"Remove empty directories. Do you wish to continue?"

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -196,10 +200,13 @@ define({root:
     Forums: "Forums",
     Forward: "Forward",
+    FoundFile2: "Found File:",
+    FoundFileMessage: "A found file has all of its parts on disk that are not referenced in the Dali server. All the file parts are accounted for so they can be added back to the Dali server. They can also be deleted from the cluster, if required.",
```
Does this text come from the legacy version?  @JamesDeFabia do you want some input on these messages?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -248,10 +255,14 @@ define({root:
     LargestFile: "Largest File",
     LargestSize: "Largest Size",
+    LastMessage: "Last Message",
+    LastMessage2: "Last Message:",
```
Could we move the ":" into the JS?

miguelvazq replied:
How about in the html I can do something like `<label class="Prompt" for="${id}LastMessage">${i18n.LastMessage2}:</label>`

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -248,10 +255,14 @@ define({root:
     LastNHours: "Last N Hours",
     LastNRows: "Last N Rows",
+    LastRun: "Last Run",
+    LastRun2: "Last Run:",
```
Could we move the ":" into the JS?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -277,6 +288,8 @@ define({root:
     Logs: "Logs",
     log_analysis_1: "log_analysis_1*",
+    LostFile2: "Lost File:",
+    LostFileMessage: "A logical file that is missing at least one file part on both the primary and replicated locations in storage. The logical file is still referenced in the Dali server. Deleting the file removes the reference from the Dali server and any remaining parts on disk.",
```
@JamesDeFabia Please review

JamesDeFabia replied:
this text is fine

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -344,6 +361,8 @@ define({root:
     Options: "Options",
     OriginalFile: "Original File",
+    OrphanFile2: "Orphan File:",
+    OrphanMessage: "An orphan file has partial file parts on disk. However, a full set of parts is not available to construct a complete logical file. These partial file parts, therefore, do not have a reference in the Dali server.",
```
@JamesDeFabia Please review

JamesDeFabia replied:
"An orphan file has partial file parts on disk. However, a full set of parts is not available to construct a complete logical file. This no reference to these file parts in the Dali server."

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -465,6 +489,7 @@ define({root:
     RowPath: "Row Path",
     RowTag: "Row Tag",
     RoxieCluster: "Roxie Cluster",
+    RunningServerStrain: "Running this process may take a long time and will put a heavy strain on the servers. Do you wish to continue?",
```
@JamesDeFabia Please Review

JamesDeFabia replied:
This text is ok

GordonSmith commented on esp/src/eclwatch/templates/XrefDetailsWidget.html:
```diff
@@ -0,0 +1,52 @@
+                    </form>
+                </div>
+            </div>
+            <div id="${id}_FoundFiles" title="Found Files" data-dojo-props="delayWidget: 'XrefFoundFilesWidget'" data-dojo-type="DelayLoadWidget">
```
translations

GordonSmith commented on esp/src/eclwatch/templates/XrefDetailsWidget.html:
```diff
@@ -0,0 +1,52 @@
+            </div>
+            <div id="${id}_FoundFiles" title="Found Files" data-dojo-props="delayWidget: 'XrefFoundFilesWidget'" data-dojo-type="DelayLoadWidget">
+            </div>
+            <div id="${id}_OrphanFiles" title="Orphan Files" data-dojo-props="delayWidget: 'XrefOrphanFilesWidget'" data-dojo-type="DelayLoadWidget">
```
translations

GordonSmith commented on esp/src/eclwatch/templates/XrefDetailsWidget.html:
```diff
@@ -0,0 +1,52 @@
+            </div>
+            <div id="${id}_OrphanFiles" title="Orphan Files" data-dojo-props="delayWidget: 'XrefOrphanFilesWidget'" data-dojo-type="DelayLoadWidget">
+            </div>
+            <div id="${id}_LostFiles" title="Lost Files" data-dojo-props="delayWidget: 'XrefLostFilesWidget'" data-dojo-type="DelayLoadWidget">
```
translations

GordonSmith commented on esp/src/eclwatch/templates/XrefDetailsWidget.html:
```diff
@@ -0,0 +1,52 @@
+            </div>
+            <div id="${id}_Directories" title="Directories" data-dojo-props="delayWidget: 'XrefDirectoriesWidget'" data-dojo-type="DelayLoadWidget">
+            </div>
+            <div id="${id}_Errors" title="Errors/Warnings" data-dojo-props="delayWidget: 'XrefErrorsWarningsWidget'" data-dojo-type="DelayLoadWidget">
```
translations

GordonSmith commented on esp/src/eclwatch/templates/XrefDetailsWidget.html:
```diff
@@ -0,0 +1,52 @@
+            </div>
+            <div id="${id}_LostFiles" title="Lost Files" data-dojo-props="delayWidget: 'XrefLostFilesWidget'" data-dojo-type="DelayLoadWidget">
+            </div>
+            <div id="${id}_Directories" title="Directories" data-dojo-props="delayWidget: 'XrefDirectoriesWidget'" data-dojo-type="DelayLoadWidget">
```
translations

GordonSmith commented on esp/src/eclwatch/templates/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,270 @@
+            this.store.getChildren = function (parent, options) {
+               var retVal =  this.query({__hpcc_parentName: parent.__hpcc_id}, options);
+               return retVal;
+            };*/
```
remove commented code.

GordonSmith replied:
Still needs removing

GordonSmith commented on esp/src/eclwatch/templates/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,270 @@
+                id: this.id + "Generate",
+                disabled: false,
+                onClick: function (val) {
+                    if (confirm("Running this process may take a long time and will put a heavy strain on the servers. Do you wish to continue?")) {
```
translation

GordonSmith replied:
Still needs translation

GordonSmith commented on esp/src/eclwatch/templates/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,270 @@
+                    context.dialog.show();
+                },
+                label: "Errors/Warnings"
+             }).placeAt(this.directories.domNode, "after");*/
```
remove commented code

GordonSmith replied:
Commented code still needs removing

GordonSmith commented on esp/src/eclwatch/templates/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,270 @@
+            new ToolbarSeparator().placeAt(this.cancel.domNode, "after");
+            /*new ToolbarSeparator().placeAt(this.errorWarnings.domNode, "after");
+            new ToolbarSeparator().placeAt(this.openButton.domNode, "after");
+            new ToolbarSeparator().placeAt(this.foundFiles.domNode, "before");*/
```
remove commented code.

GordonSmith replied:
Needs removing

GordonSmith commented on esp/src/eclwatch/templates/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,270 @@
+            return retVal;
+            return {
+                Name: "SuperFiles"
+            }*/
```
remove commented code (is filter used in this widget?)

GordonSmith replied:
commented code

GordonSmith commented on esp/src/eclwatch/templates/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,270 @@
+                    });
+                });
+                context.store.setData(newRows);
+                context.grid.set("query", /*this.getFilter()*/{});
```
remove commented code (and all refs to filters?)

GordonSmith replied:
Commented code.

GordonSmith commented on esp/src/eclwatch/templates/XrefErrorsWarningsWidget.js:
```diff
@@ -0,0 +1,270 @@
+            id = this.createChildTabID(id);
+            var retVal = registry.byId(id);
+            if (!retVal) {
+                var context = this;
```
context not used.

GordonSmith replied:
Context can be removed?

GordonSmith commented on esp/src/eclwatch/templates/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,270 @@
+            this.store.getChildren = function (parent, options) {
+               var retVal =  this.query({__hpcc_parentName: parent.__hpcc_id}, options);
+               return retVal;
+            };*/
```
remove commented code

GordonSmith commented on esp/src/eclwatch/templates/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,270 @@
+                id: this.id + "Generate",
+                disabled: false,
+                onClick: function (val) {
+                    if (confirm("Running this process may take a long time and will put a heavy strain on the servers. Do you wish to continue?")) {
```
translation

GordonSmith commented on esp/src/eclwatch/templates/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,270 @@
+                    context.dialog.show();
+                },
+                label: "Errors/Warnings"
+             }).placeAt(this.directories.domNode, "after");*/
```
remove commented code.

GordonSmith commented on esp/src/eclwatch/templates/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,270 @@
+            new ToolbarSeparator().placeAt(this.cancel.domNode, "after");
+            /*new ToolbarSeparator().placeAt(this.errorWarnings.domNode, "after");
+            new ToolbarSeparator().placeAt(this.openButton.domNode, "after");
+            new ToolbarSeparator().placeAt(this.foundFiles.domNode, "before");*/
```
remove commented code.

GordonSmith commented on esp/src/eclwatch/templates/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,270 @@
+            return retVal;
+            return {
+                Name: "SuperFiles"
+            }*/
```
remove commented code.

GordonSmith commented on esp/src/eclwatch/templates/XrefLostFilesWidget.js:
```diff
@@ -0,0 +1,270 @@
+                    });
+                });
+                context.store.setData(newRows);
+                context.grid.set("query", /*this.getFilter()*/{});
```
remove commentedd code and all refs to filters?

miguelvazq commented on esp/src/eclwatch/XrefDetailsWidget.js:
```diff
@@ -0,0 +1,177 @@
+
+        initTab: function () {
+            var currSel = this.getSelectedChild();
+            if (currSel.id == this.widget._FoundFiles.id && !this.widget._FoundFiles.__hpcc_initalized) {
```
I started seeing those in your code review and the `for (key` will now == `for (var key`

richardkchapman commented:
@miguelvazq Back to you

miguelvazq commented:
@GordonSmith Updated

HPCCSmoketest commented:
Automated Smoketest 
Sha: b7dd3691f87e6593d99006b4c9f0d6c59e8dd402
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -196,10 +201,14 @@ define({root:
     Forums: "Forums",
     Forward: "Forward",
+    FoundFile: "Found Files",
+    FoundFile2: "Found File:",
```
This can be removed

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -277,6 +290,9 @@ define({root:
     Logs: "Logs",
     log_analysis_1: "log_analysis_1*",
+    LostFile: "Lost Files:",
+    LostFile2: "Lost File:",
```
Are the ":" going to be moved into the HTML?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -344,6 +364,9 @@ define({root:
     Options: "Options",
     OriginalFile: "Original File",
+    OrphanFile: "Orphan Files:",
+    OrphanFile2: "Orphan File:",
```
Are the ":" going to be moved into the HTML?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -556,12 +587,15 @@ define({root:
     title_AvailableGroupPermissions: "Available Group Permissions",
     title_ClusterInfo: "Groups",
     title_CodeGeneratorPermissions: "Code Generator Permissions",
+    title_DirectoriesFor: "Directories for:",
```
Are the ":" going to be moved into the HTML?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -556,12 +587,15 @@ define({root:
     title_DFUWUDetails: "DFU Workunit",
     title_DiskUsage: "Disk Usage",
     title_ECLPlayground: "ECL Playground",
+    title_ErrorsWarnings: "Errors/Warnings for:",
```
Are the ":" going to be moved into the HTML?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -556,12 +587,15 @@ define({root:
+    title_ErrorsWarnings: "Errors/Warnings for:",
     title_EventScheduleWorkunit: "Event Scheduler",
     title_FileScopeDefaultPermissions: "Default permissions of files",
+    title_FoundFilesFor: "Found files for:",
```
Are the ":" going to be moved into the HTML?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -579,7 +613,9 @@ define({root:
     title_LZBrowse: "Landing Zones",
     title_MemberOf: "Member Of",
     title_Members: "Members",
+    title_LostFilesFor: "Lost files for:",
```
Are the ":" going to be moved into the HTML?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -579,7 +613,9 @@ define({root:
     title_Members: "Members",
+    title_LostFilesFor: "Lost files for:",
     title_LibrariesUsed: "Libraries Used",
+    title_OrphanFilesFor: "Orphan files for:",
```
Are the ":" going to be moved into the HTML?

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -248,10 +257,14 @@ define({root:
     LargestFile: "Largest File",
     LargestSize: "Largest Size",
+    LastMessage: "Last Message",
+    LastMessage2: "Last Message",
```
LastMessage2 Not needed (same as LastMessage)

GordonSmith commented on esp/src/eclwatch/nls/hpcc.js:
```diff
@@ -248,10 +257,14 @@ define({root:
     LastNHours: "Last N Hours",
     LastNRows: "Last N Rows",
+    LastRun: "Last Run",
+    LastRun2: "Last Run",
```
LastRun2 duplicate of LastRun

richardkchapman commented:
@miguelvazq Back to you

miguelvazq commented:
@GordonSmith updated.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 0849c7dbf32f0f3e8bc5216b335ca802a782b8e9
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: f6f50a979b4107697d401380f59d15c2749c04bc
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -0,0 +1,99 @@
+                    Name: {label: this.i18n.Name, width:100, sortable: false},
+                    Modified: {label: this.i18n.Modified, width: 30, sortable: true},
+                    Parts: {label: this.i18n.Parts, width: 30, sortable: true},
+                    Size: {label: this.i18n.Size, width: 30, sortable: true},
```
Trailing ","

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -0,0 +1,130 @@
+            this.refreshGrid();
+        },
+
+        createGrid: function (domID) {\
```
trailing "\"

GordonSmith commented on esp/src/eclwatch/XrefQueryWidget.js:
```diff
@@ -0,0 +1,216 @@
+                        }
+                    },
+                    Modified: {label: this.i18n.LastRun, width: 30, sortable: false},
+                    Status: {label: this.i18n.LastMessage, width: 30, sortable: false},
```
trailing ","

Commit:
HPCC-12131 Xref Upgrade

Incorporate Xref into the new ecl watch paradigm.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

miguelvazq commented:
I removed an accidental file that was added into templates. The ones with the commented code comments. All else was updated.

HPCCSmoketest commented:
Automated Smoketest 
Sha: bc843747f2866f60d5365edd68547ea6f299374f
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: 5a0fb3bccbbe5731ecbacb1d8fa131e25ee51d1a
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK
